tests/scripts/test_bump_version.py

```
@@ -0,0 +1,522 @@
+        assert v.python() == "0.9.0.post2"
+        assert v.semver() == "0.9.0-post2"
+        assert v.dashed() == "0-9-0-post2"
+        assert v.is_post
+
+    def test_parse_post_dash(self, bv):
+        v = bv.Version.parse("0.9.0-post2")
+        assert v == bv.Version.parse("0.9.0.post2")
+
+    @pytest.mark.parametrize("bad", ["1.2", "1.2.3.4", "v1.2.3", "1.2.3-rc1", "abc"])
+    def test_parse_rejects_invalid(self, bv, bad):
+        with pytest.raises(Exception):
```

Ruff (0.15.10)

[warning] Do not assert blind exception: `Exception`

(B017)